apps/forklift/lib/forklift/event/event_handler.ex

```diff
@@ -21,16 +22,17 @@ defmodule Forklift.Event.EventHandler do
 
   def handle_event(%Brook.Event{
         type: data_ingest_start(),
-        data: %Dataset{technical: %{sourceType: type}} = dataset,
+        data: %Ingestion{targetDataset: dataset_id} = ingestion,
         author: author
-      })
-      when type in ["stream", "ingest"] do
+      }) do
     data_ingest_start()
-    |> add_event_count(author, dataset.id)
+    |> add_event_count(author, dataset_id)
 
-    :ok = Forklift.DataReaderHelper.init(dataset)
+    dataset = Forklift.Datasets.get!(dataset_id)
 
-    Forklift.Datasets.update(dataset)
+    if dataset != nil do
+      :ok = Forklift.DataReaderHelper.init(dataset)
+    end
```

Comment:
can we add a more explicit return in the event that the dataset is nil? Thinking about implicit returns and how previously, we were previously returning the result of `Forklift.Datasets.update(dataset)`

Reply:
Good catch! Added an explicit :ok to the end of the event handler function.